src/components/VmActions/AutoSelectConsole.js

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState } from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import { addUserMessage, downloadVmConsoles } from '_/actions'
+import ConsoleConfirmationModal from '_/components/VmActions/ConsoleConfirmationModal'
+import { msg } from '_/intl'
+
+const AutoSelectConsole = (props) => {
+  const {
+    vm,
+    show,
+    defaultConsole,
+    isNoVNC,
+    fetchConsoles,
+    requestActive,
+    id,
+    onFailure,
+  } = props
+
+  const initialState = {
+    // consoles: [],
+    selectedProtocol: undefined,
+    failureReason: undefined,
+    dispatchStatus: 'doNotDispatch',
+    active: false,
+  }
+
+  const [state, setState] = useState(initialState)
+  const {
+    dispatchStatus,
+    selectedProtocol,
+    active,
+    // failureReason,
+    // consoles,
+  } = state
+
+  useEffect(() => {
+    if (show) {
+      setState(state => ({
+        ...state,
+        dispatchStatus: 'shouldDispatch',
+        active: true,
+      }))
+      fetchConsoles(vm.get('id'))
+    }
+  }, [show]) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
IMO the vm, the fetchConsole shouldn't trigger the function since nothing happened unless show === true.
so why it should be in the dependency list? vm might change on any refresh but it shouldn't affect the AutoSelectConsole component. WDYT?

Reply:
OK. The we should provide VM ID as dependency(i.e. via props or local variable).
```....
    fetchConsoles(vmId)}}, [show, vmId])
```